### PR TITLE
Move _drop_rows_by_label from Frame to IndexedFrame

### PR DESCRIPTION
**Pull request by vyasr**
This PR is a trivial move of a function from frame.py to indexed_frame.py. That function is only used in one place throughout cuDF, inside `IndexedFrame.drop`.

### CODE REVIEW

**Review by brandon-b-miller — APPROVED** [review]

**Review by bdice — APPROVED** [review]
Approving because it's a trivial move -- but I have two suggestions for fixes/improvement.

File: python/cudf/cudf/core/indexed_frame.py [hunk 1]
```diff
@@ -4553,3 +4553,84 @@ def _is_series(obj):
...
+
+
+@_cudf_nvtx_annotate
+def _drop_rows_by_labels(
+    obj: DataFrameOrSeries,
+    labels: Union[ColumnLike, abc.Iterable, str],
+    level: Union[int, str],
+    errors: str,
+) -> DataFrameOrSeries:
+    """Remove rows specified by `labels`. If `errors=True`, an error is raised
```
Comment (vyasr) [inline python/cudf/cudf/core/indexed_frame.py:4565]:
```suggestion
    """Remove rows specified by `labels`. If `errors="raise"`, an error is raised
```

**Comment by vyasr** [comment]
rerun tests

**Comment by bdice** [comment]
@gpucibot merge

**Pull request merged**